### ISSUE TITLE
Shurikens are supposed to pierce

### ISSUE THREAD

**Issue by BlackVegetable** [akintu/akintu#239]
I'm putting this issue on your plate until you inform me of the best way to implement this.  At that point, feel free to assign it to me.

Shuriken are supposed to pierce through enemies in a line to hit the next enemy some of the time.  This should happen with every physical attack with a shuriken.  I think you have a method that would be perfect for this, but let me know how you think this should be done.

This is ultra-low priority, as the only class that really shines with the shuriken will not be implemented by release.

**Comment by malikoth** [comment]
It's also really easy.

First, we borrow combat.py line 500 from the knockback method, which calculates the end of a line drawn from the target location in the same direction as the path from the source to the target.  Then we pass that in as the 'end' argument to getLineTargets.  So, like this:

newloc = target.cLocation.move(source.cLocation.direction_to(target.cLocation), distance)
targets = getLineTargets(cPane, target.cLocation, newloc)

This will return a list of Person objects with at least one Person in it, since that's the person who is getting hit in the first place.  Then just check that len(targets) >= 2 to make sure there's at least one more target, and apply the secondary damage to targets[1].  There may be yet more Persons in that list, but you only want a shuriken to pierce one target for a maximum number of possible targets of 2, right?  So if there are more in that list, ignore them.

Also, you said "Supposed to hit the next enemy some of the time" and "Should happen with every physical attack with a shuriken"  So which is it?  Some of the time, or every physical attack?

**Comment by BlackVegetable** [comment]
Both.  There is a 1/3 chance of a first pierce, a 1/9 chance of the second pierce, and a 1/27 chance of a third pierce.  Every physical shuriken attack will need this.  But yeah.

**Comment by malikoth** [comment]
OK, so:

```
chance = 1.0
for p in targets:
  if random.random() <= chance:
    #Apply damage to p
  else:
    break
  chance /= 3
```

Do you need anything else from me on this?

**Comment by BlackVegetable** [comment]
Nah, I just wanted your input on which region-ish method to use.  Thanks, I'll take it from here.
